src/doc_builder/autodoc.py

```diff
@@ -463,7 +476,10 @@ def _resolve_link(search):
         if anchor not in mapping:
             return f"`{object_name}`"
         page = f"{prefix}{mapping[anchor]}"
-        return f"[{object_name}]({page}#{anchor}){last_char}"
+        if "#" in page:
+            return f"[{object_name}]({page}){last_char}"
+        else:
+            return f"[{object_name}]({page}#{anchor}){last_char}"
```

Comment:
Some of the pages already have the anchor in the mapping (the ones where the object name is different from the anchor name).